tools/doc_examples.py

```diff
@@ -0,0 +1,172 @@
+        target_path.write_text(code)
+        print(f"Extracted example to: {target_path}")  # noqa: T201
+
+        if generate_tests:
+            test_file_path.write_text(
+                _generate_test_code(
+                    module_name=str(target_path.with_suffix("")).replace("/", "."),
+                    name=target_path.stem,
+                    code=test_file_path.read_text() if i else "",
+                    counter=i,
+                )
+            )
+            print(f"Created test stub in: {test_file_path}")  # noqa: T201
+
+    file.write_text(content, encoding="utf-8")
+
+
+def check_docs(max_line_length: int = 15) -> int:
+    """Check docs for inline code examples that should be separate files."""
+    errors = 0
+    for file in Path("docs").rglob("*.md"):
+        content = file.read_text()
+        content_lines = content.splitlines()
+        if DISABLE_EXAMPLE_LINT.search(content_lines[0]):
+            continue
+
+        for code_block in _find_code_blocks(content, max_line_length):
+            errors += 1
+            lineno = _find_matching_lineno(content_lines, code_block)
+            print(  # noqa: T201
+                f"{file}:{lineno}: Inline code examples longer than 15 lines are not allowed "
+                "and should be put in a separate file"
+            )
+    return errors
+
+
+def check_command(args: Namespace) -> None:
+    """Run `check_docs` with args from the CLI."""
+    raise SystemExit(1 if check_docs(max_line_length=args.line_length) else 0)
+
+
+def extract_command(args: Namespace) -> None:
+    """Run `extract_examples` with args from the CLI."""
+    extract_examples(
+        file_name=args.filename,
+        target_dir_name=args.directory,
+        name=args.name,
+        max_line_length=args.line_length,
+    )
+
+
+def cli() -> None:
+    """Convenience CLI."""
+    parser = ArgumentParser()
+    subparsers = parser.add_subparsers()
+    parser_check = subparsers.add_parser("check", help="check all doc files")
+    parser_extract = subparsers.add_parser("extract", help="extract inline code examples from a file")
+
+    parser_check.add_argument("--line-length", default=15, help="maximum line length for inline code blocks")
+
+    parser_extract.add_argument("filename", help="source file name")
+    parser_extract.add_argument("-n", "--name", help="name of the generated module without extension")
+    parser_extract.add_argument("-d", "--directory", default="examples", help="target directory for extracted files")
+    parser_extract.add_argument("-l", "--line-length", default=15, help="maximum line length for inline code blocks")
+    parser_extract.add_argument("-t", "--test-stub", action="store_true", default=True, help="generate test stubs")
+
+    parser_check.set_defaults(func=check_command)
+    parser_extract.set_defaults(func=extract_command)
+
+    args = parser.parse_args()
+    args.func(args)
+
+
+if __name__ == "__main__":
+    cli()
```

Comment:
can you document this cli ?

Reply:
WDYM? Or rather, where do you think it should be documented?